instrumentation/src/androidTest/java/com/bumptech/glide/DarkModeTest.java

```diff
@@ -51,6 +52,45 @@ public void before() {
     assumeTrue(VERSION.SDK_INT >= VERSION_CODES.Q);
   }
 
+  @Test
+  public void load_withDarkModeActivity_vectorDrawable_usesDarkModeColor() {
+    try (ActivityScenario<FragmentActivity> scenario = darkModeActivity()) {
+      scenario.onActivity(
+          activity -> {
+            ViewGroup container = findContainer(activity);
+            ImageView imageView = newFixedSizeImageView(activity);
+            container.addView(imageView);
+
+            Glide.with(activity)
+                .load(R.drawable.vector_drawable)
+                .override(Target.SIZE_ORIGINAL)
+                .into(imageView);
+          });
+
+      onIdle();
+      scenario.onActivity(
+          activity -> {
+            ViewGroup container = findContainer(activity);
+            ImageView imageView = (ImageView) container.getChildAt(0);
+            Bitmap result = drawToBitmap(imageView.getDrawable());
+            Bitmap expected = drawToBitmap(activity.getDrawable(R.drawable.vector_drawable_dark));
+            assertThat(result).sameAs(expected);
+          });
+    }
+  }
```

Comment:
**Optimization Issue**: The method 'load_withDarkModeActivity_vectorDrawable_usesDarkModeColor' performs a drawable to bitmap conversion inside the activity's onActivity method. This process is not optimized for performance. Converting a drawable to a bitmap is a CPU-intensive operation and doing it on the main thread can lead to frame drops, especially if the drawable is complex or the device is low-end. <br> **Fix**: To optimize this process, consider moving the drawable to bitmap conversion into a background thread or using Glide's built-in asynchronous loading and decoding capabilities. Glide already has efficient mechanisms for loading and decoding images in the background, which can significantly improve the performance of image loading and rendering. <br> **Code Suggestion**: 
 ```
 +  new Thread(() -> {
 +    Bitmap result = drawToBitmap(imageView.getDrawable());
 +    Bitmap expected = drawToBitmap(activity.getDrawable(R.drawable.vector_drawable_dark));
 +    runOnUiThread(() -> assertThat(result).sameAs(expected));
 +  }).start();
 ```